### ISSUE TITLE
Change Command: remove_equip

### ISSUE THREAD

**Issue by DestroyedClone** [harbingerofme/DebugToolkit#111]
**Which Command**
`remove_equip`

**What should it do?**
Equipment slots can have up to 256 equipment stored, though this is only put into effect by MUL-T with 2 equipment slots (the second equipment isn't cleared upon switching characters, such as in cases with Artifact of Metamorphosis and transforming into Heretic).
For debugging, an additional slot argument could be included in order to remove equipment from these slots.

**Comment by SChinchi** [comment]
Just asking, do you think there may be any debugging reasons to do that?

The extra equipment persists because `Inventory.equipmentStateSlots` is never shortened, only extended. So even if you switch to MUL-T's alt and give it an equipment and then remove it, the array is not resized down and the alt equipment display window will stay. The only way to make that display go away is by resizing the equipment state array.

The alt equipment persisting when changing bodies may look like a bug, but it's just the game displaying the data, even if inaccessible. And by keeping it there, you are able to access it again when you switch back to MUL-T.

There may be a sequence breaking reason when testing some mod where clearing the array makes sense. An idea would be in `remove_equip` to add a bit of logic after removing the equipment from the current slot to iterate the equipment state array backwards and resize it down to the latest slot that doesn't have `EquipmentState.empty`. However, there would still be an edge case:
- have both equipment slots populated with MUL-T and set slot 0 as the current one
- switch to another body and call `remove_equip`
- Slot 1 is still populated and the state array won't be resized down

My opinion at the moment is that all of this is unnecessary.

**Comment by harbingerofme** [comment]
I don't think we necessarily need to resize the array, just remove all equipment from each slot.

**Comment by SChinchi** [comment]
Closing this because as of Alloyed Collective the secondary equipment icon is used to display other equipment in the same slot with the Functional Coupler. As such, MUL-T's alternative equipment is not an issue anymore and if this functionality is ever restored, at least we'll have `remove_equip_extra` (c23fb5f).
